libraries/AP_Filesystem/AP_Filesystem.cpp

```diff
@@ -286,6 +287,37 @@ bool AP_Filesystem::fgets(char *buf, uint8_t buflen, int fd)
     return true;
 }
 
+// run crc32 over file with given name, returns true if successful
+bool AP_Filesystem::crc32(const char *fname, uint32_t& checksum)
+{
+    // Open file in readonly mode
+    int fd = open(fname, O_RDONLY);
+    if (fd == -1) {
+        return false;
+    }
+
+    // Buffer to store data temporarily
+    const ssize_t buff_len = 64;
+    uint8_t buf[buff_len];
```

Comment:
In the case of the GCS FTP were now using a second buffer rather than re-using the FTP one. We could also pass in a buffer and size I guess....

Reply:
We decided this was OK on the dev all.